spec/rubocop/cop/style/redundant_parentheses_spec.rb

```diff
@@ -765,6 +765,45 @@ def foo
     RUBY
   end
 
+  it 'registers parentheses around `&&` logical operator keywords when chained to `&&` operator' do
+    expect_offense(<<~RUBY)
+      (x && y) && z
+      ^^^^^^^^ Don't use parentheses around a logical expression.
+
+      x && (y && z)
+           ^^^^^^^^ Don't use parentheses around a logical expression.
+
+      x && (y && z) && w
+           ^^^^^^^^ Don't use parentheses around a logical expression.
+
+      x && (y && z) || w
+           ^^^^^^^^ Don't use parentheses around a logical expression.
+
+      x || (y && z) && w
+           ^^^^^^^^ Don't use parentheses around a logical expression.
```

Comment:
Can you separate the test cases as much as possible?

Reply:
Done